File: 240408/총점 비교/compare-total-points.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.Arrays;

class Student implements Comparable<Student> {

    String name;
    int kor, eng, math;

    public Student(String name, int kor, int eng, int math) {
        this.name = name;
        this.kor = kor;
        this.eng = eng;
        this.math = math;
    }

    public int scoreSum() {
        return kor + eng + math;
    }

     @Override
    public int compareTo(Student other) {
        return this.scoreSum() - other.scoreSum();
    }
};


public class Main {

    static int N;
    static StringTokenizer st;
    static Student[] students;
    static StringBuilder result = new StringBuilder();

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        students = new Student[N];
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            students[i] = new Student(
                st.nextToken(),
                Integer.parseInt(st.nextToken()), 
                Integer.parseInt(st.nextToken()), 
                Integer.parseInt(st.nextToken())
            );
        }

        Arrays.sort(students);
        for (Student s : students) {
            result.append(String.format("%s %d %d %d\n", s.name, s.kor, s.eng, s.math));
        }

        System.out.print(result);
    }
}

File: 240417/총점 비교/compare-total-points.py
class Student:
    def __init__(self, n, a, b, c):
        self.n = n
        self.a = a
        self.b = b
        self.c = c

N = int(input())
students = []
for _ in range(N):
    n, a, b, c = input().split()
    students.append(Student(n, int(a), int(b), int(c)))

students.sort(key = lambda s: (s.a + s.b +s.c))
for s in students:
    print(s.n, s.a, s.b, s.c)